mne/minimum_norm/inverse.py

```diff
@@ -1332,8 +1332,152 @@ def _prepare_forward(forward, info, noise_cov, pca=False, rank=None,
+
+    if (depth is not None or loose < 1) and not (forward['surf_ori'] or
+                                                 allow_fixed_depth):
+        logger.info('Forward is not surface oriented, converting.')
+        forward = convert_forward_solution(
+            forward, surf_ori=True, use_cps=use_cps)
+
+    gain_info, gain = _select_orient_forward(forward, info, noise_cov)
+    noise_cov = prepare_noise_cov(noise_cov, info, gain_info['ch_names'], rank)
+    whitener, _ = compute_whitener(
+        noise_cov, info, gain_info['ch_names'], pca=pca)
+    logger.info("Selected %d channels" % (len(gain_info['ch_names'],)))
+    assert whiten in ('before', 'after')
+    if whiten == 'before':
+        logger.info('Whitening the forward solution.')
+        gain = np.dot(whitener, gain)
+
+    if depth is None:
+        depth_prior = None
+    else:
+        patch_areas = forward.get('patch_areas', None)
+        depth_prior = compute_depth_prior(
+            gain, gain_info, is_fixed_ori, exp=depth, patch_areas=patch_areas,
+            limit_depth_chs=limit_depth_chs, loose_method=loose_method,
+            limit=limit)
+
+    # Deal with fixed orientation forward / inverse
+    if fixed:
+        orient_prior = None
+        if not is_fixed_ori:
+            # Convert to the fixed orientation forward solution now
+            if depth is not None:
+                # Convert the depth prior into a fixed-orientation one
+                logger.info('    Picked elements from a free-orientation '
+                            'depth-weighting prior into the fixed-orientation '
+                            'one')
+                depth_prior = depth_prior[2::3]
+        if not allow_fixed_depth:
+            forward = convert_forward_solution(
+                forward, surf_ori=forward['surf_ori'], force_fixed=True,
+                use_cps=use_cps)
+            gain_info, gain = _select_orient_forward(
+                forward, info, noise_cov, verbose=False)
+    else:
+        # In theory we could have orient_prior=None for loose=1., but
+        # the MNE-C code does not do this
+        orient_prior = compute_orient_prior(forward, loose=loose)
+    if whiten == 'after':
+        logger.info('Whitening the forward solution.')
+        gain = np.dot(whitener, gain)
+
+    logger.info('Creating the source covariance matrix')
+    source_std = np.ones(gain.shape[1])
+    if depth_prior is not None:
+        source_std *= depth_prior
+    if orient_prior is not None:
+        source_std *= orient_prior
+    np.sqrt(source_std, out=source_std)
+    gain *= source_std
+    # Adjusting Source Covariance matrix to make trace of G*R*G' equal
+    # to number of sensors.
+    logger.info('Adjusting source covariance matrix.')
+    trace_GRGT = linalg.norm(gain, ord='fro') ** 2
+    n_nzero = (noise_cov['eig'] > 0).sum()
+    scale = np.sqrt(n_nzero / trace_GRGT)
+    source_std *= scale
+    gain *= scale
```

Comment:
this big block above is just a copy / paste?

Reply:
Yes the math in `minimum_norm` should not have changed (it was quite sensitive to changes because we benchmark all `inv` keys to MNE-C)